dsa-solutions/lc-solutions/0000-0099/0098-validate-binary-search-tree.md

```diff
@@ -0,0 +1,95 @@
+---
+id: validate binary search tree
+title: Validate Binary Search Tree
+sidebar_label: 0098 Validate Binary Search Tree
+tags:
+  - tree
+  - tree traversal
+  - LeetCode
+  - C++
+description: "This is a solution to the Validate Binary Search Tree problem on LeetCode."
+---
+
+## Problem Description
+
+Given the root of a binary tree, determine if it is a valid binary search tree (BST).
+
+A valid BST is defined as follows:
+
+* The left subtree of a node contains only nodes with keys less than the node's key.
+* The right subtree of a node contains only nodes with keys greater than the node's key.
+* Both the left and right subtrees must also be binary search trees.
+
+### Examples
+
+**Example 1:**
+
+```
+
+Input: root = [2,1,3]
+Output:  true
+```
+
+**Example 2:**
+
+```
+Input: root = [5,1,4,null,null,3,6]
+Output: false
+```
+
+### Constraints
+
+- The number of nodes in the tree is in the range $[1, 10^4]$.
+- -2^(31) <= Node.val <= 2^(31) - 1.
```

Comment:
replace `- -2^(31) <= Node.val <= 2^(31) - 1.` to `- $-2^(31) <= Node.val <= 2^(31) - 1$`.